### ISSUE TITLE
[Bug]: Chat doesn't seem to have access to the code

### ISSUE THREAD

**Issue by grangej** [intitni/CopilotForXcode#385]
### Before Reporting

- [X] I have checked FAQ, and there is no solution to my issue
- [X] I have searched the existing issues, and there is no existing issue for my issue

### What happened?

Recently the chat no longer has access to the code, even when I manually specify the @code in the prompt.

For example: 

Error:  

I apologize for the inconvenience, but it seems there is a technical issue. The updated code from the file "/*****/features/DraftInvite/DraftInviteView.swift" was not loaded correctly. The context still shows an empty file. 

Could you please provide the updated code directly or describe the changes you've made so I can assist you more effectively?

### How to reproduce the bug.

Make changes to the code and ask for updates to the code

### macOS version

14.2

### Xcode version

15.0.1

### Copilot for Xcode version

0.26

### Node version

v20.8.0

**Comment by intitni** [comment]
It seems to work for me.

<img width="1192" alt="Screenshot 2023-11-05 at 12 36 35" src="https://github.com/intitni/CopilotForXcode/assets/793147/32aa17c5-2b90-4e2a-adef-dce5f1fcb361">

Please let me know if you were selecting empty spaces in the editor like this?

<img width="499" alt="Screenshot 2023-11-05 at 12 38 42" src="https://github.com/intitni/CopilotForXcode/assets/793147/ff66c9a4-e7a6-4828-9824-71463c1690ed">

Does it work if you select a piece of code and ask about it?

Do you think you can MITM the request and see if the code is contained in the request body?

```json
{
  "messages": [
    {
      "content": "You must always reply in Chinese\nYou are an AI programming assistant.\nYour reply should be concise, clear, informative and logical.\nYou MUST reply in the format of markdown.\nYou MUST embed every code you provide in a markdown code block.\nYou MUST add the programming language name at the start of the markdown code block.\nIf you are asked to help perform a task, you MUST think step-by-step, then describe each step concisely.\nIf you are asked to explain code, if it's too long, you SHOULD explain it step-by-step in a ordered list, concisely.\nMake your answer short and structured.\n ## System Info\n\nCurrent Time: Sunday, 2023-11-05 12:36:18+0800 (You can use it to calculate time in another time zone)\n\n## File and Code Scope\n\nYou can use the following context to answer my questions about the editing document or code. The context shows only a part of the code in the editing document, and will change during the conversation, so it may not match our conversation.\n\nWhen you don't known what I am asking, I am probably referring to the code.\n\n### Editing Document Context\n\nDocument Relative Path: 
<--The Code is Inserted Here Inside Content !-->
\/Tool\/Sources\/ChatContextCollectors\/ActiveDocumentChatContextCollector\/ActiveDocumentChatContextCollector.swift\n\nLanguage: swift\n\nFocused Context:\n```\npublic final class ActiveDocumentChatContextCollector: ChatContextCollector\nfunc getActiveDocumentContext(_ info: EditorInformation) -> ActiveDocumentContext\n```\n\nFocused Range [line, character]: [76, 8] - [86, 9]\n\nFocused Code (start from line 76):\n```swift\n        var activeDocumentContext = activeDocumentContext ?? .init(\n            filePath: \"\",\n            relativePath: \"\",\n            language: .builtIn(.swift),\n            fileContent: \"\",\n            lines: [],\n            selectedCode: \"\",\n            selectionRange: .outOfScope,\n            lineAnnotations: [],\n            imports: []\n        )\n\n```",
      "role": "system"
    },
    {
      "content": "what does this function do? (reply in English)",
      "role": "user"
    }
  ],
  "functions": [
    {
      "description": "Call when Editing Document Context provides too little context to answer a question.",
      "name": "expandFocusRange",
      "parameters": {
        "type": "object",
        "properties": {
        }
      }
    }
  ],
  "model": "gpt-35-turbo",
  "temperature": 0.7000000000000001,
  "stream": true
}
```

**Comment by grangej** [comment]
Unfortunately it does seem to be working this morning.

I will try and reproduce again and get a screenshot of both the editor and chat window as you have done. I did also at the time make sure that I wasn't selecting anything and also even tried selecting everything in the editing file. Perhaps it is the size of the file?

Happy to also try and proxy the request once I can reproduce it again.

**Comment by intitni** [comment]
One possible reason is that the app failed to find the UI element for the editor, I will review the logic before the next release.

Currently, the app will try to find the element each time Xcode becomes active or the focused UI element changes (switching between editors or windows, etc.)

If you encounter this issue again, you can try deactivating and reactivating Xcode as a workaround.

**Comment by intitni** [comment]
Released in 0.27.0
